lib/screens/admin_order_detail_page.dart

```diff
@@ -0,0 +1,689 @@
+                ),
+              ),
+            ],
+          )),
+    );
+  }
+
+  Widget _buildCustomerDetails() {
+    return Container(
+      decoration: BoxDecoration(
+        color: kPaletteWhite,
+        borderRadius: BorderRadius.only(
+          topLeft: Radius.circular(40),
+          topRight: Radius.circular(40),
+        ),
+      ),
+      child: StreamBuilder<DocumentSnapshot>(
+          stream: Firestore.instance
+              .collection(constants.DBCollections.users)
+              .document(widget.order.customerId)
+              .snapshots(),
+          builder: (context, snapshot) {
+            if (!snapshot.hasData) {
+              return LinearProgressIndicator();
+            }
+            Customer _customer = Customer.fromSnapshot(snapshot.data);
+            return Column(
+              children: <Widget>[
+                _buildGreyBar(),
+                _buildCustomerContactInfoBox(_customer),
+                _buildShippingAddressInfoBox(_customer),
+                _buildShippingMethodInfoBox(),
+              ],
+            );
+          }),
+    );
+  }
+
+  Widget _buildGreyBar() {
+    return Container(
+      padding: EdgeInsets.only(top: 24.0),
+      child: Container(
+        decoration: BoxDecoration(
+          color: paletteGreyColor4,
+          borderRadius: BorderRadius.all(Radius.circular(40)),
+        ),
+        height: 8,
+        width: 115,
+      ),
+    );
+  }
+
+  Widget _buildCustomerContactInfoBox(Customer customer) {
+    return Container(
+      padding:
+          EdgeInsets.only(top: 12.0, left: 24.0, right: 24.0, bottom: 36.0),
+      child: Column(
+        children: <Widget>[
+          Container(
+            alignment: Alignment.center,
+            child: Text("注文者情報",
+                style: TextStyle(
+                  fontSize: 18,
+                  color: paletteBlackColor,
+                  fontWeight: FontWeight.bold,
+                )),
+          ),
+          SizedBox(height: 24.0),
+          _buildCustomerName(customer),
+          SizedBox(height: 12.0),
+          _buildCustomerAddress(customer),
+          SizedBox(height: 6.0),
+          _buildCustomerPhoneNumber(customer),
+          SizedBox(height: 6.0),
+          _buildCustomerEmailAddress(customer),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildCustomerName(Customer customer) {
+    return Container(
+      alignment: Alignment.centerLeft,
+      child: Row(
+        children: <Widget>[
+          Text(customer.name,
+              style: TextStyle(
+                fontSize: 16,
+                color: paletteBlackColor,
+                fontWeight: FontWeight.w600,
+              )),
+          Text(" 様",
+              style: TextStyle(
+                fontSize: 16,
+                color: paletteBlackColor,
+                fontWeight: FontWeight.w300,
+              )),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildCustomerAddress(Customer customer) {
```

Comment:
I think there’s some code duplication between customer address and shipping address. Can we do something to reduce the duplication?

Reply:
@dragonfly91 
You are right, indeed there is duplication!
I added a `Widget _buildAddressRow(Address address)` and removed the duplication.